src/main/resources/public/ts/controllers/eval_teacher_ctl.ts

```diff
@@ -3204,15 +3204,27 @@ export let evaluationsController = ng.controller('EvaluationsController', [
          * Calcul la moyenne pour un élève
          * @param eleve élève
          */
-        $scope.calculerMoyenneEleve = function (eleve) {
-            eleve.getMoyenne($scope.releveNote.devoirs.all).then(() => {
-                if(!eleve.moyenneFinaleIsSet){
-                    eleve.moyenneFinale = eleve.moyenne;
-                    eleve.oldMoyenneFinale = eleve.moyenne;
-                }
-                $scope.calculerMoyenneClasse();
-                utils.safeApply($scope);
-            });
+        $scope.calculerMoyenneEleve = function (eleve, idClasse, idMatiere?) {
+            if (idMatiere != null) {
+                eleve.getMoyenne($scope.releveNote.devoirs.all, $scope.releveNote.idEtablissement, idClasse, $scope.releveNote.periode.id, idMatiere).then(() => {
+                    if(!eleve.moyenneFinaleIsSet){
+                        eleve.moyenneFinale = eleve.moyenne;
+                        eleve.oldMoyenneFinale = eleve.moyenne;
+                    }
+                    $scope.calculerMoyenneClasse();
+                    utils.safeApply($scope);
+                });
+            }
+            else {
+                eleve.getMoyenne($scope.releveNote.devoirs.all).then(() => {
+                    if(!eleve.moyenneFinaleIsSet){
+                        eleve.moyenneFinale = eleve.moyenne;
+                        eleve.oldMoyenneFinale = eleve.moyenne;
+                    }
+                    $scope.calculerMoyenneClasse();
+                    utils.safeApply($scope);
+                });
+            }
```

Comment:
```suggestion
            let promise: Promise<any>
            if (idMatiere != null) {
                promise = eleve.getMoyenne($scope.releveNote.devoirs.all, $scope.releveNote.idEtablissement, idClasse, $scope.releveNote.periode.id, idMatiere)
            }
            else {
                promise = eleve.getMoyenne($scope.releveNote.devoirs.all)
            }
            promise.then(() => {
                    if(!eleve.moyenneFinaleIsSet){
                        eleve.moyenneFinale = eleve.moyenne;
                        eleve.oldMoyenneFinale = eleve.moyenne;
                    }
                    $scope.calculerMoyenneClasse();
                    utils.safeApply($scope);
                });
```
Je ne sais pas si ce que je propose marche, a tester.

Reply:
Il peut faire direct en await

et faire sa verification directement

```suggestion
     if (idMatiere != null) {
                await eleve.getMoyenne($scope.releveNote.devoirs.all, $scope.releveNote.idEtablissement, idClasse, $scope.releveNote.periode.id, idMatiere);
            } else {
                await eleve.getMoyenne($scope.releveNote.devoirs.all);
            }
            if(!eleve.moyenneFinaleIsSet){
                eleve.moyenneFinale = eleve.moyenne;
                eleve.oldMoyenneFinale = eleve.moyenne;
            }
            $scope.calculerMoyenneClasse();
            utils.safeApply($scope);
```

j'ai pas mis de try catch car je sais pas ce qu'il se passe si ça fail ?